test/externalTests.sh

```diff
@@ -28,84 +28,160 @@
 
 set -e
 
+if [ "$CIRCLECI" ]
+then
+    function printTask() { echo ""; echo "$(tput bold)$(tput setaf 2)$1$(tput setaf 7)"; }
+    function printError() { echo ""; echo "$(tput setaf 1)$1$(tput setaf 7)"; }
+else
+    function printTask() { echo ""; echo "$(tput bold)$(tput setaf 2)$1$(tput sgr0)"; }
+    function printError() { echo ""; echo "$(tput setaf 1)$1$(tput sgr0)"; }
+fi
+
 if [ ! -f "$1" ]
 then
   echo "Usage: $0 <path to soljson.js>"
   exit 1
 fi
 
 SOLJSON="$1"
+SOLCVERSION="UNDEFINED"
+
+function setup_solcjs
+{
+  cd "$1"
+  git clone --depth 1 -b v0.5.0 https://github.com/ethereum/solc-js.git solc
+
+  cd solc
+  npm install
+  cp "$SOLJSON" soljson.js
+  SOLCVERSION=$(./solcjs --version)
+  cd ..
+  echo "Using solcjs version $SOLCVERSION"
+}
+
+function download_project
+{
+  local repo="$1"
+  local branch="$2"
+  local dir="$3"
+
+  if [ -n "$branch" ]
```

Comment:
If you pass a dir but not a branch, won't it use the dir as branch?

Reply:
Oh, nice catch! Wanted to have the order similar for all function calls, which is why I put `$dir` to the end. I'll make the `$branch` non-optional since the projects we test do not have a identical branch names anyways.